File: backend/DrugInteractionAPI/Models/CheckResult.cs
using System.ComponentModel.DataAnnotations;

namespace DrugInteractionAPI.Models;

public class CheckResult
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();
    
    [Required]
    public Guid JobId { get; set; }
    
    public string Summary { get; set; } = string.Empty;
    
    [Required]
    public string Categories { get; set; } = string.Empty; // JSON: {danger, caution, recommendation}
    
    public double? ConfidenceScore { get; set; }
    
    public string? Sources { get; set; } // JSON array of sources
    
    public string? RawModelOutput { get; set; } // Redacted OpenAI response
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    // Navigation property
    public CheckJob Job { get; set; } = null!;
}

public class CheckResultResponse
{
    public Guid JobId { get; set; }
    public JobStatus Status { get; set; }
    public string? Summary { get; set; }
    public InteractionCategories? Categories { get; set; }
    public double? ConfidenceScore { get; set; }
    public List<SourceReference>? Sources { get; set; }
}

public class InteractionCategories
{
    public List<DrugInteraction> Danger { get; set; } = new();
    public List<DrugInteraction> Caution { get; set; } = new();
    public List<DrugInteraction> Recommendation { get; set; } = new();
}

public class DrugInteraction
{
    public List<string> Meds { get; set; } = new();
    public string Category { get; set; } = string.Empty;
    public string ShortSummary { get; set; } = string.Empty;
    public string Mechanism { get; set; } = string.Empty;
    public List<string> SymptomsToMonitor { get; set; } = new();
    public List<EvidenceItem> Evidence { get; set; } = new();
    public string SuggestedAction { get; set; } = string.Empty;
    public double Confidence { get; set; }
}

public class EvidenceItem
{
    public string Source { get; set; } = string.Empty;
    public string Id { get; set; } = string.Empty;
    public string Quote { get; set; } = string.Empty;
}

public class SourceReference
{
    public string Type { get; set; } = string.Empty;
    public string Url { get; set; } = string.Empty;
}

File: mobile/lib/models/check_result.dart
import 'package:json_annotation/json_annotation.dart';

part 'check_result.g.dart';

@JsonSerializable()
class CheckResult {
  @JsonKey(name: 'job_id')
  final String jobId;
  final String status;
  final String? summary;
  final InteractionCategories? categories;
  @JsonKey(name: 'confidence_score')
  final double? confidenceScore;
  final List<SourceReference>? sources;

  CheckResult({
    required this.jobId,
    required this.status,
    this.summary,
    this.categories,
    this.confidenceScore,
    this.sources,
  });

  factory CheckResult.fromJson(Map<String, dynamic> json) => _$CheckResultFromJson(json);
  Map<String, dynamic> toJson() => _$CheckResultToJson(this);
}

@JsonSerializable()
class InteractionCategories {
  final List<DrugInteraction> danger;
  final List<DrugInteraction> caution;
  final List<DrugInteraction> recommendation;

  InteractionCategories({
    this.danger = const [],
    this.caution = const [],
    this.recommendation = const [],
  });

  factory InteractionCategories.fromJson(Map<String, dynamic> json) => _$InteractionCategoriesFromJson(json);
  Map<String, dynamic> toJson() => _$InteractionCategoriesToJson(this);
}

@JsonSerializable()
class DrugInteraction {
  final List<String> meds;
  final String category;
  @JsonKey(name: 'short_summary')
  final String shortSummary;
  final String mechanism;
  @JsonKey(name: 'symptoms_to_monitor')
  final List<String> symptomsToMonitor;
  final List<EvidenceItem> evidence;
  @JsonKey(name: 'suggested_action')
  final String suggestedAction;
  final double confidence;

  DrugInteraction({
    required this.meds,
    required this.category,
    required this.shortSummary,
    required this.mechanism,
    this.symptomsToMonitor = const [],
    this.evidence = const [],
    required this.suggestedAction,
    required this.confidence,
  });

  factory DrugInteraction.fromJson(Map<String, dynamic> json) => _$DrugInteractionFromJson(json);
  Map<String, dynamic> toJson() => _$DrugInteractionToJson(this);
}

@JsonSerializable()
class EvidenceItem {
  final String source;
  final String id;
  final String quote;

  EvidenceItem({
    required this.source,
    required this.id,
    required this.quote,
  });

  factory EvidenceItem.fromJson(Map<String, dynamic> json) => _$EvidenceItemFromJson(json);
  Map<String, dynamic> toJson() => _$EvidenceItemToJson(this);
}

@JsonSerializable()
class SourceReference {
  final String type;
  final String url;

  SourceReference({
    required this.type,
    required this.url,
  });

  factory SourceReference.fromJson(Map<String, dynamic> json) => _$SourceReferenceFromJson(json);
  Map<String, dynamic> toJson() => _$SourceReferenceToJson(this);
}